src/cli/README_HISTORY.md

```diff
@@ -0,0 +1,290 @@
+
+Timestamps use millisecond accuracy and are tacked up to 49 days. If the event is older than 49 days, the entry is still tracked in the list but the timestamp is shown as `more than 49 days ago`.
+
+## Command Details
+
+### help
+
+Usage: `history help`
+
+Print SRP client help menu.
+
+```bash
+> history help
+help
+netinfo
+rx
+tx
+Done
+>
+```
+
+### netinfo
+
+Usage `history netinfo [list] [<num-entries>]`
+
+Print the Network Info history. Each Network Info provides
+
+- Device Role
+- MLE Link Mode
+- RLOC16
+- Partition ID
+
+Print the Network Info history as a table.
+
+```bash
+> history netinfo
+| Time                     | Role     | Mode | RLOC16 | Partition ID |
++--------------------------+----------+------+--------+--------------+
+|         00:00:10.069 ago | router   | rdn  | 0x6000 |    151029327 |
+|         00:02:09.337 ago | child    | rdn  | 0x2001 |    151029327 |
+|         00:02:09.338 ago | child    | rdn  | 0x2001 |    151029327 |
+|         00:07:40.806 ago | child    | -    | 0x2001 |    151029327 |
+|         00:07:42.297 ago | detached | -    | 0x6000 |            0 |
+|         00:07:42.968 ago | disabled | -    | 0x6000 |            0 |
+Done
+```
+
+Print the Network Info history as a list.
+
+```bash
+> history netinfo list
+00:00:59.467 ago -> role:router mode:rdn rloc16:0x6000 partition-id:151029327
+00:02:58.735 ago -> role:child mode:rdn rloc16:0x2001 partition-id:151029327
+00:02:58.736 ago -> role:child mode:rdn rloc16:0x2001 partition-id:151029327
+00:08:30.204 ago -> role:child mode:- rloc16:0x2001 partition-id:151029327
+00:08:31.695 ago -> role:detached mode:- rloc16:0x6000 partition-id:0
+00:08:32.366 ago -> role:disabled mode:- rloc16:0x6000 partition-id:0
+Done
+```
+
+Print only the latest 2 entries.
+
+```bash
+> history netinfo 2
+| Time                     | Role     | Mode | RLOC16 | Partition ID |
++--------------------------+----------+------+--------+--------------+
+|         00:02:05.451 ago | router   | rdn  | 0x6000 |    151029327 |
+|         00:04:04.719 ago | child    | rdn  | 0x2001 |    151029327 |
+Done
+```
+
+### rx
+
+Usage `history rx [list] [<num-entries>]`
+
+Print the IPv6 message RX history. Each entry provides:
+
+- IPv6 message type: UDP, TCP, ICMP6 (and its subtype), etc.
+- IPv6 payload length (excludes the IPv6 header).
+- Source IPv6 address and port number.
+- Destination IPv6 address and port number (port number is valid for UDP/TCP, it is zero otherwise).
+- Whether or not link-layer security was used.
+- Message priority: low, norm, high, net (for Thread Control traffic).
+- RSS: Received Signal Strength (in dBm) - averaged over all received fragment frames that formed the message.
+- Short address (RLOC16) of neighbor from which message was received (`0xfffe` if not known).
+- Radio link on which the message was received (useful when `OPENTHREAD_CONFIG_MULTI_RADIO` is enabled).
+
+Print the IPv6 message RX history as a table:
+
+```bash
+> history rx
```

Comment:
What about filtering the messages by its type? For example, if I care only about UDP messages, I can use `history rx udp`.

Reply:
These are good suggestions.  Thanks.

This is the initial implementation of the `HistoryTracker` feature (adding its foundations). My intention is that we build on top of this in future PRs.

There are a lot of other ideas/functionality we can later add onto this. I was considering some of the same things that you suggested, my idea is to later add a general filter model: filter based on type, or msg size, or src or dst address, whether it used link security, priority, etc. and/or allow user to define a combination of conditions to filter.

There are also some ideas on new items/entries to have `HistoryTracker` track:
- Child table and neighbor table (keep history when a child/neighbor attaches/detaches, etc).
- History of changes to network data (when prefix/route/service entries are added/removed, etc).
- History of unicast/multicast IPv6 addresses 
- History of route table changes (next hops, router table, router ids, track network topology).
- ...

I suggest leaving more functions/feature for future PRs.